src/topology/sheaves/presheaf.lean

```diff
@@ -44,6 +44,41 @@ variables {C}
 
 namespace presheaf
 
+local attribute [instance] concrete_category.has_coe_to_sort concrete_category.has_coe_to_fun
+
+/-- The restriction of a section along an inclusion of open sets.
+For `x : F.obj (op V)`, we provide the notation `x ∣_ₕ i` (`h` stands for `hom`) for `i : U ⟶ V`,
+and the notation `x ∣_ₗ U ⟪i⟫` (`l` stands for `le`) for `i : U ≤ V`.
+-/
+def restrict {X : Top} {C : Type*} [category C] [concrete_category C]
+  {F : X.presheaf C} {V : opens X} (x : F.obj (op V)) {U : opens X} (h : U ⟶ V) : F.obj (op U) :=
+F.map h.op x
+
+infixl ` ∣_ₕ `: 80 := restrict
+
+notation x ` ∣_ₗ `: 80 U ` ⟪` e `⟫ ` := @restrict _ _ _ _ _ _ x U
+  (@hom_of_le (opens _) _ U _ e)
```

Comment:
What's the reason for including `U` here? It won't elaborate without `U`?
```suggestion
notation x ` ∣_ₗ `: 80 U ` ⟪` e `⟫ ` := @restrict _ _ _ _ _ _ x U (@hom_of_le (opens _) _ U _ e)
```

Reply:
It looks more alike to what one would write on paper, it shows up on the info view, and sometimes we would need to specify `U` by type annotations on `e` nonetheless, say when `e = le_top`. Plus it is always possible to omit it via an underscore.

I'm not so sure with the notation of the `⟪e⟫` part though.